spec/models/printables/group_spec.rb

```diff
@@ -82,75 +61,142 @@
+      it 'sends the right message to PMB' do
+        request =
+          stub_request(:post, "#{uri}/print_jobs")
+            .with(
+              body: {
+                data: {
+                  type: 'print_jobs',
+                  attributes: {
+                    printer_name: 'printer1',
+                    label_template_id: 1,
+                    labels: {
+                      body: [
+                        { label: { barcode: '2', barcode2d: '2', top_line: '', middle_line: nil, bottom_line: '' } },
+                        { label: { barcode: '1', barcode2d: '1', top_line: '', middle_line: nil, bottom_line: '' } }
+                      ]
+                    }
+                  }
+                }
+              }.to_json,
+              headers: post_headers
+            )
+            .to_return(status: 200, body: <<~RESPONSE, headers: { 'Content-Type' => 'application/json' })
+                {"data":{"type":"print_jobs","attributes":{"printer_name":"printer1","label_template_id":1,"labels":{"body":[{"label":{"barcode":"2","barcode2d":"2","top_line":"","middle_line":null,"bottom_line":""}},{"label":{"barcode":"1","barcode2d":"1","top_line":"","middle_line":null,"bottom_line":""}}]}}}}
+                RESPONSE
+
+        group.print(config, 'user1')
+        expect(request).to have_been_made
+      end
+
+      context 'when an asset does not have barcode' do
+        let(:asset1) { create(:asset, facts: facts1, barcode: nil) }
+        it 'does not print it' do
+          request =
+            stub_request(:post, "#{uri}/print_jobs")
+              .with(
+                body: {
+                  data: {
+                    type: 'print_jobs',
+                    attributes: {
+                      printer_name: 'printer1',
+                      label_template_id: 1,
+                      labels: {
+                        body: [
+                          { label: { barcode: '2', barcode2d: '2', top_line: '', middle_line: nil, bottom_line: '' } }
+                        ]
+                      }
+                    }
+                  }
+                }.to_json,
+                headers: post_headers
+              )
+              .to_return(status: 200, body: <<~RESPONSE, headers: { 'Content-Type' => 'application/vnd.api+json' })
+                  {"data":{"type":"print_jobs","attributes":{"printer_name":"printer1","label_template_id":1,"labels":{"body":[{"label":{"barcode":"1","barcode2d":"1","top_line":"","middle_line":null,"bottom_line":""}}]}}}}
+                  RESPONSE
+
+          group.print(config, 'user1')
+          expect(request).to have_been_made
+        end
+      end
+    end
+
+    context 'when using v2' do
+      let(:uri) { 'http://localhost:10000/v2' }
+      let(:post_headers) { { 'Content-Type' => 'application/json' } }
+      xit 'sends the right message to PMB' do
```

Comment:
Does this want to be x-ed out?

Reply:
No... no we do not. I could swear I had those running...

Reply:
Okay, they are okay locally... I'll have to double check I haven't done anything silly with the commit range